index.js

```diff
@@ -153,8 +168,16 @@ NYC.prototype._wrapRequire = function () {
 
     _this.sourceMapCache.add(filename, code)
 
-    // now instrument the compiled code.
-    return _this.instrumenter.instrumentSync(code, './' + relFile)
+    var hash = md5(code)
+    var cacheFilePath = path.join(_this._cacheDirectory(), hash + '.js')
+
+    try {
+      return fs.readFileSync(cacheFilePath, 'utf8')
+    } catch (e) {
+      var instrumented = _this.instrumenter().instrumentSync(code, './' + relFile)
+      fs.writeFile(cacheFilePath, instrumented)
```

Comment:
I suppose it's a cache so you don't need to block the process nor do you care if it fails… maybe add a comment to that effect?